### PR TITLE
Move oximeter-collector timeseries definitions to TOML

### PR DESCRIPTION
**Pull request by bnaecker**
Also fixup some optional deps and innards in `oximeter-db` based on the selected features

### CODE REVIEW

File: oximeter/collector/src/self_stats.rs [hunk 1]
```diff
@@ -9,51 +9,21 @@
...
 use oximeter::Sample;
-use oximeter::Target;
 use reqwest::StatusCode;
 use std::borrow::Cow;
 use std::collections::BTreeMap;
-use std::net::IpAddr;
 use std::time::Duration;
-use uuid::Uuid;
+
+oximeter::use_timeseries!("oximeter-collector.toml");
```
Comment (smklein) [inline oximeter/collector/src/self_stats.rs:19]: Should this file exist in the PR? Or is it getting auto-generated in some way I'm not seeing?

Comment (bnaecker) [reply oximeter/collector/src/self_stats.rs:19]: It should, if I remembered to add it :) Thanks, adding now.

Comment (bnaecker) [reply oximeter/collector/src/self_stats.rs:19]: Added in f101ca03b

File: oximeter/collector/src/self_stats.rs [hunk 2]
```diff
@@ -9,51 +9,21 @@
...
-    /// The producer's IP address.
-    pub producer_ip: IpAddr,
-    /// The producer's port.
-    pub producer_port: u16,
-    /// The base route in the producer server used to collect metrics.
-    ///
-    /// The full route is `{base_route}/{producer_id}`.
-    ///
-    // TODO-cleanup: This is no longer relevant, but removing it entirely
-    // relies on nonexistent functionality for updating timeseries schema. When
```
Comment (smklein) [inline oximeter/collector/src/self_stats.rs:51]: Should this TODO be copied over? Same for `FailedCollections`

**Review by smklein — APPROVED** [review]

File: oximeter/collector/src/self_stats.rs (hunk 2 above)
Comment (bnaecker) [reply oximeter/collector/src/self_stats.rs:51]: Good question, I made https://github.com/oxidecomputer/omicron/issues/6028 to keep track of that. Thanks!

**Comment by bnaecker** [comment]
Waiting to coordinate with @zeeshanlakhani on the feature-twiddling.

**Comment by bnaecker** [comment]
I rebased on the feature-flag bits in a9024e8d7ff30f1ef3ef9aad83d9e71a059ef5af

**Pull request merged**